### ISSUE TITLE
cannot build latest stable version on Python 3.12

### ISSUE THREAD

**Issue by Gui-FernandesBR** [uber/h3-py#326]
Python 3.12 was recently release, but apparently there's a problem with h3's wheel on PyPI.

I'm working on a project that depends on h3, therefore I cannot build my packge unless I drop this dependency.

This is the full traceback (more than 300 lines so I attached as .txt): [error_29_oct.txt](https://github.com/uber/h3-py/files/13198671/error_29_oct.txt)

And this is the short version: 

```txt
Collecting h3
  Using cached h3-3.7.6.tar.gz (178 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: h3
  Building wheel for h3 (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for h3 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [356 lines of output]

....
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for h3
Failed to build h3
ERROR: Could not build wheels for h3, which is required to install pyproject.toml-based projects
```

My environment has these specifications:
`Python 3.12.0 (tags/v3.12.0:0fb18b0, Oct  2 2023, 13:03:39) [MSC v.1935 64 bit (AMD64)] on win32`

**Comment by jannikmi** [comment]
@Gui-FernandesBR I could install version 3.7.6 with python 3.12. Do you still get the error?

**Comment by Gui-FernandesBR** [comment]
> @Gui-FernandesBR I could install version 3.7.6 with python 3.12. Do you still get the error?

Yes it is running now. I think something was fixed in the python `3.12.1` version.
